docs/architecture.md

```diff
@@ -0,0 +1,216 @@
+                                    |                                       |    |                                    |
+                                    |    Microsoft.Git.CredentialManager    <----+ Microsoft.Git.CredentialManager.UI |
+                                    |                                       |    |                                    |
+                                    +---------------------------------------+    +------------------------------------+
+```
+
+Git Credential Manager Core (GCM Core) is built to be Git host and platform/OS agonstic.
+Most of the shared logic (command execution, the abstract platform subsystems, etc) can be found in the
+`Microsoft.Git.CredentialManager` class library (C#). The library targets .NET Standard as well as .NET Framework.
+
+> **Note**
+>
+> The reason for also targeting .NET Framework directly is that the `Microsoft.Identity.Client`
+> ([MSAL.NET](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet)) library requires a .NET Framework
+> target to be able to show the embedded web browser auth pop-up on Windows platforms.
+>
+> There are extension points that now exist in MSAL.NET meaning we can plug-in our own browser pop-up handling code
+> on .NET Core meaning both Windows and Mac. We haven't yet gotten around to exploring this.
+
+The entry-point for GCM Core can be found in the `Git-Credential-Manager` project, a console application that targets both
+.NET Core and .NET Framework. This project emits the `git-credential-manager-core(.exe)` executable, and contains very little
+code - registration of all supported host providers and running the `Application` object found in `Microsoft.Git.CredentialManager`.
+
+Providers have their own projects/assemblies that take dependencies on the `Microsoft.Git.CredentialManager` core assembly,
+and are dependents of the main entry point application `Git-Credential-Manager`. Code in these binaries is expected to run
+on all supported platforms and typically (see MSAL.NET note above) does not include any graphical user interface; they use
+terminal prompts only.
+
+Where a provider needs some platform-specific interaction or graphical user interface, the recommended model is to have a
+separate 'helper' executable that the shared, core binaries shell out to. Currently the Bitbucket and GitHub providers each
+have a WPF (Windows only) helper executable that shows authentication prompts and messages.
+
+The `Microsoft.Git.CredentialHelper.UI` project is a WPF (Windows only) assembly that contains common WPF components and styles
+that are shared between provider helpers on Windows.
+
+### Cross-platform UI
+
+We hope to be able to migrate the WPF/Windows only helpers to [Avalonia](https://avaloniaui.net/) in order to gain cross-platform
+graphical user interface support. See [this](https://github.com/microsoft/Git-Credential-Manager-Core/issues/136) issue for
```

Comment:
Instead of "See [this] issue" could it just be "See #136"? I forget if GitHub's markdown works that way. Of course, that won't work when viewing this file in VS Code or something.

Reply:
I'd prefer to avoid the `#136` style here in documentation I think.